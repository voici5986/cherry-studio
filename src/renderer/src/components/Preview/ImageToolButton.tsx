import { Button, Tooltip } from 'antd'
import { memo } from 'react'

interface ImageToolButtonProps {
  tooltip: string
  icon: React.ReactNode
  onClick: () => void
}

const ImageToolButton = ({ tooltip, icon, onClick }: ImageToolButtonProps) => {
  return (
    <Tooltip title={tooltip} mouseEnterDelay={0.5} mouseLeaveDelay={0}>
      <Button shape="circle" icon={icon} onClick={onClick} role="button" aria-label={tooltip} />
    </Tooltip>
  )
}

export default memo(ImageToolButton)
