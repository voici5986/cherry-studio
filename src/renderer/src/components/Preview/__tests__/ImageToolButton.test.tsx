import { render } from '@testing-library/react'
import { beforeEach, describe, expect, it, vi } from 'vitest'

import ImageToolButton from '../ImageToolButton'

// Mock antd components
vi.mock('antd', () => ({
  Button: vi.fn(({ children, onClick, ...props }) => (
    <button type="button" data-testid="custom-button" onClick={onClick} {...props}>
      {children}
    </button>
  )),
  Tooltip: vi.fn(({ children, title }) => <div title={title}>{children}</div>)
}))

describe('ImageToolButton', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  const defaultProps = {
    tooltip: 'Test tooltip',
    icon: <span data-testid="test-icon">Icon</span>,
    onClick: vi.fn()
  }

  it('should match snapshot', () => {
    const { asFragment } = render(<ImageToolButton {...defaultProps} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
