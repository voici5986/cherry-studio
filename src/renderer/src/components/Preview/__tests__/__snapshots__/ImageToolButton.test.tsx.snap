// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ImageToolButton > should match snapshot 1`] = `
<DocumentFragment>
  <div
    title="Test tooltip"
  >
    <button
      aria-label="Test tooltip"
      data-testid="custom-button"
      icon="[object Object]"
      role="button"
      shape="circle"
      type="button"
    />
  </div>
</DocumentFragment>
`;
